RetroAchievements/RAWeb
# refactor(head-analytics): migrate php code to an action, add coverage

This PR migrates the URL path sanitization and props extraction code in _head-analytics.blade.php_ to a reusable action that is also covered by tests.

Currently, there is an issue in Plausible where our self-healing system URLs are not being sanitized correctly:
![Screenshot 2024-12-23 at 1 25 35 PM](https://github.com/user-attachments/assets/50cd96cb-fae3-41c8-b7c3-df527756f901)

This will be a recurring problem going forward as more URLs use slugs. The changes in this PR address the issue:
![Screenshot 2024-12-23 at 1 26 25 PM](https://github.com/user-attachments/assets/472f1a55-8862-4e4c-9f4a-8daa9d2ea40f)

**How to test this PR**
Open your network tab and navigate to various pages. Check the "event" call. The `p` and `u` parameters are the ones of interest.
